app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -2,62 +2,142 @@
 
 import com.example.mycalculator.mvp.contract.CalculatorContract;
 
+import java.util.function.Function;
+
+import static com.example.mycalculator.utils.StringUtils.ERROR_MESSAGE_DIVISION;
+import static com.example.mycalculator.utils.StringUtils.ERROR_MESSAGE_INVALID_FORMAT;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_DIVIDE;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_MINUS;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_MULTIPLY;
 import static com.example.mycalculator.utils.StringUtils.OPERATOR_SUM;
 import static com.example.mycalculator.utils.StringUtils.EMPTY_STRING;
 import static com.example.mycalculator.utils.StringUtils.ERROR_MESSAGE;
+import static com.example.mycalculator.utils.StringUtils.TEXT_ZERO_NUMBER;
 
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand = EMPTY_STRING;
-    private String secondOperand = EMPTY_STRING;
+    private Operand firstOperand = new Operand();
+    private Operand secondOperand = new Operand();
     private String operator = EMPTY_STRING;
     private String result = EMPTY_STRING;
 
+    private class Operand {
```

Comment:
You shouldnt create an inner class, you could create this one in a separete file

Reply:
you should create it in a separete class and inject them in the constructor, if not you would not be able to test

Reply:
Done, class another created.